engenere_partner_sales_info/models/res_partner.py

```diff
@@ -0,0 +1,334 @@
+    )
+    last_open_quotation_id = fields.Many2one(
+        "sale.order",
+        string="Last Open Quotation",
+        compute="_compute_open_quotation_info",
+    )
+    last_open_quotation_date = fields.Date(
+        string="Last Quotation Date",
+        compute="_compute_open_quotation_info",
+        help="Date of the most recent draft/sent quotation",
+    )
+
+    # ========== COTAÇÕES ==========
+
+    def _compute_open_quotation_info(self):
+        """Calcula cotações em aberto (draft/sent)"""
+        for partner in self:
+            partner.has_open_quotation = False
+            partner.last_open_quotation_id = False
+            partner.last_open_quotation_date = False
+
+        customer_partners = self.filtered(lambda p: p.customer_rank > 0)
+        if not customer_partners:
+            return
+
+        open_quotations = self.env["sale.order"].search(
+            [
+                ("partner_id", "in", customer_partners.ids),
+                ("state", "in", ["draft", "sent"]),
+            ],
+            order="date_order DESC, id DESC",
+        )
+        partner_quotations = {}
+        for quot in open_quotations:
+            if quot.partner_id.id not in partner_quotations:
+                partner_quotations[quot.partner_id.id] = quot
+
+        for partner in customer_partners:
+            if partner.id in partner_quotations:
+                last_quot = partner_quotations[partner.id]
+                partner.has_open_quotation = True
+                partner.last_open_quotation_id = last_quot.id
+                if last_quot.date_order:
+                    partner.last_open_quotation_date = last_quot.date_order.date()
+
+    # ========== MENSAGEM ==========
+
+    def _compute_analysis_message(self):
+        config_param = self.env["ir.config_parameter"].sudo()
+        months = config_param.get_param(
+            "engenere_partner_sales_info.default_analysis_months", 24
+        )
+        analysis_months = int(months)
+        message = _(
+            "Analysis period: %(months)d months. "
+            "Note: bonus amounts are not excluded."
+        ) % {"months": analysis_months}
+        for partner in self:
+            partner.analysis_message = message
+
+    # ========== FUNÇÕES AUXILIARES ==========
+
+    def _get_analysis_months(self):
+        config_param = self.env["ir.config_parameter"].sudo()
+        return int(
+            config_param.get_param(
+                "engenere_partner_sales_info.default_analysis_months", 24
+            )
+        )
+
+    def _get_start_date(self, analysis_months):
+        return fields.Date.context_today(self) - relativedelta(months=analysis_months)
+
+    def _group_records_by_partner(self, records):
+        grouped = defaultdict(list)
+        for record in records:
+            grouped[record.partner_id.id].append(record)
+        return grouped
+
+    def _compute_record_stats(self, records, date_extractor, amount_extractor):
```

Comment:
```suggestion
    def _prepare_record_statistics_vals(self, records, date_extractor, amount_extractor):
```
o nome do método podia ser assim, já que essa função não é um compute de fato, utilizado diretamente em um campo.